ext/afform/core/Civi/Api4/Action/Afform/SubmitFile.php

```diff
@@ -98,15 +98,23 @@ protected function processForm() {
 
     $file = civicrm_api3('Attachment', 'create', $attachmentParams);
 
-    // Update multi-record custom field with value
     if (strpos($apiEntity, 'Custom_') === 0) {
+      // Update multi-record custom field with value
       civicrm_api4($apiEntity, 'update', [
         'values' => [
           $idField => $entityId,
           $this->fieldName => $file['id'],
         ],
       ]);
     }
+    else {
+      // Update entity field
+      $api4 = $this->_formDataModel->getSecureApi4($this->modelName);
+      $api4($apiEntity, 'update', [
+        'values' => [$this->fieldName => $file['id']],
+        'where' => [[$idField, '=', $entityId]],
+      ]);
+    }
```

Comment:
It seems these 2 blocks of code are mostly the same and could be consolidated.

FYI, there is a reason to use `civicrm_api` like the top block instead of `getSecureApi4` like the bottom block:
The latter restricts actions, so if the form is configured to allow "create" but not "update" this will fail, even though in this case the "update" should be allowed because it is just a supplementary addition to the original "create".

Reply:
Changed that. In case the security model is set to `FBAC` the callable returned by `getSecureApi4` executes the action with `checkPermissions` set to `FALSE`. Wouldn't the same be necessary here?

Reply:
Yes I agree it's fine to set it to false.

Reply:
`checkPermisisons` is now set to `FALSE`.